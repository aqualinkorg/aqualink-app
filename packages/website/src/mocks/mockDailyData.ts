import { DailyData } from "../store/Reefs/types";

const now = new Date();
const minutesAgo = 5;
const dailyDataDate = new Date(
  now.getTime() - minutesAgo * 60000
).toISOString();

export const mockDailyData: DailyData = {
  id: 1,
  date: dailyDataDate,
  minBottomTemperature: 25,
  maxBottomTemperature: 27,
  avgBottomTemperature: 26,
  degreeHeatingDays: 29,
  surfaceTemperature: 36,
  satelliteTemperature: 20,
  minWaveHeight: 1,
  maxWaveHeight: 3,
  avgWaveHeight: 2,
  waveDirection: 136,
  wavePeriod: 15,
  minWindSpeed: 3,
  maxWindSpeed: 5,
  avgWindSpeed: 4,
  windDirection: 96,
  weeklyAlertLevel: 3,
};
