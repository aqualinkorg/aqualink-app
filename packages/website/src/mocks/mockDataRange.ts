import { TimeSeriesDataRange } from 'store/Sites/types';

export const mockDataRange: TimeSeriesDataRange = {
  bottomTemperature: [
    {
      type: 'hobo',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  cholorophyllConcentration: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  odoConcentration: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  ph: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
  salinity: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
    {
      type: 'hui',
      depth: null,
      data: [
        {
          minDate: '2022-06-11T00:00:00.000Z',
          maxDate: '2022-06-14T23:45:00.000Z',
        },
      ],
    },
  ],
  turbidity: [
    {
      type: 'sonde',
      depth: null,
      data: [
        {
          minDate: '2017-11-02T15:00:00.000Z',
          maxDate: '2020-02-02T21:00:00.000Z',
        },
      ],
    },
  ],
};
