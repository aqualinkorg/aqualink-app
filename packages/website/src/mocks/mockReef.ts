import { CollectionMetrics, Point, Reef } from "../store/Reefs/types";
import { mockUser } from "./mockUser";

const now = new Date();
const minutesAgo = 5;
const liveDataDate = new Date(now.getTime() - minutesAgo * 60000).toISOString();

export const mockReef: Reef = {
  id: 1,
  name: "",
  polygon: {
    coordinates: [0, 0],
    type: "Point",
  },
  hasHobo: false,
  spotterId: null,
  maxMonthlyMean: 0,
  depth: 0,
  status: "in_review",
  videoStream: null,
  region: { name: "Hawaii" },
  admins: [mockUser],
  surveyPoints: [],
  stream: null,
  historicalMonthlyMean: [],
  liveData: {
    reef: { id: 1 },
    bottomTemperature: {
      value: 39,
      timestamp: liveDataDate,
    },
    satelliteTemperature: {
      value: 29,
      timestamp: liveDataDate,
    },
    degreeHeatingDays: {
      value: 34,
      timestamp: liveDataDate,
    },
    waveHeight: {
      value: 1,
      timestamp: liveDataDate,
    },
    waveDirection: {
      value: 90,
      timestamp: liveDataDate,
    },
    wavePeriod: {
      value: 3,
      timestamp: liveDataDate,
    },
    windSpeed: {
      value: 6,
      timestamp: liveDataDate,
    },
    windDirection: {
      value: 180,
      timestamp: liveDataDate,
    },
  },
  dailyData: [
    {
      id: 171,
      date: liveDataDate,
      minBottomTemperature: 37,
      maxBottomTemperature: 39,
      avgBottomTemperature: 38,
      degreeHeatingDays: 34,
      topTemperature: 29,
      satelliteTemperature: 23,
      minWaveHeight: 2,
      maxWaveHeight: 4,
      avgWaveHeight: 3,
      waveDirection: 205,
      wavePeriod: 28,
      minWindSpeed: 3,
      maxWindSpeed: 5,
      avgWindSpeed: 4,
      windDirection: 229,
    },
  ],
  latestDailyData: {
    id: 171,
    date: liveDataDate,
    minBottomTemperature: 37,
    maxBottomTemperature: 39,
    avgBottomTemperature: 38,
    degreeHeatingDays: 34,
    topTemperature: 29,
    satelliteTemperature: 23,
    minWaveHeight: 2,
    maxWaveHeight: 4,
    avgWaveHeight: 3,
    waveDirection: 205,
    wavePeriod: 28,
    minWindSpeed: 3,
    maxWindSpeed: 5,
    avgWindSpeed: 4,
    windDirection: 229,
  },
  collectionData: {
    topTemperature: 9.54,
    bottomTemperature: 10.719999999999999,
    satelliteTemperature: 15.8900003433228,
    degreeHeatingDays: 0,
    weeklyAlertLevel: 0,
    sstAnomaly: 1.2,
  },
};

export const generateMockReef = ({
  id,
  name,
  regionName,
  deployed,
  isCollection,
  historicMax,
  collectionData,
  polygon,
}: {
  id: number;
  name: string;
  regionName: string;
  deployed: boolean;
  isCollection: boolean;
  historicMax: number;
  collectionData: CollectionMetrics;
  polygon: Point;
}): Reef => ({
  ...mockReef,
  id,
  name,
  region: { ...mockReef.region, name: regionName },
  polygon,
  status: deployed ? "deployed" : mockReef.status,
  spotterId: deployed ? "SPOT-0930" : mockReef.spotterId,
  collectionData,
  latestDailyData: isCollection ? undefined : mockReef.latestDailyData,
  maxMonthlyMean: historicMax,
});
