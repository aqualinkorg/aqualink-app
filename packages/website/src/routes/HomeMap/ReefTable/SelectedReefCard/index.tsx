/* eslint-disable no-nested-ternary */
import React from "react";
import { isNumber } from "lodash";
import {
  Box,
  Button,
  Card,
  CardMedia,
  CircularProgress,
  Grid,
  Typography,
  Hidden,
  useTheme,
  useMediaQuery,
} from "@material-ui/core";
import LaunchIcon from "@material-ui/icons/Launch";
import { Link } from "react-router-dom";

import { useSelector } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import { sortByDate } from "../../../../helpers/sortDailyData";
import { formatNumber } from "../../../../helpers/numberUtils";

import { degreeHeatingWeeksCalculator } from "../../../../helpers/degreeHeatingWeeks";
import {
  reefDetailsSelector,
  reefLoadingSelector,
} from "../../../../store/Reefs/selectedReefSlice";
import { getReefNameAndRegion } from "../../../../store/Reefs/helpers";
import { Reef } from "../../../../store/Reefs/types";
import Chart from "../../../../common/Chart";
import { surveyListSelector } from "../../../../store/Survey/surveyListSlice";
import { convertDailyDataToLocalTime } from "../../../../helpers/dates";
import Chip from "../../../../common/Chip";

const useStyles = makeStyles((theme) => ({
  cardWrapper: {
    minHeight: "18rem",
    [theme.breakpoints.down("md")]: {
      minHeight: "22rem",
    },
  },
  mobileCardWrapperWithImage: {
    [theme.breakpoints.down("xs")]: {
      height: "42rem",
    },
  },
  mobileCardWrapperWithNoImage: {
    [theme.breakpoints.down("xs")]: {
      height: "27rem",
    },
  },
  card: {
    [theme.breakpoints.down("xs")]: {
      padding: 10,
    },
    padding: 20,
  },
  cardImage: {
    borderRadius: "4px 0 0 4px",
    height: "100%",

    [theme.breakpoints.down("xs")]: {
      height: 300,
    },
  },
  metricsContainer: {
    display: "flex",
    justifyContent: "space-around",
    alignItems: "center",
    flexDirection: "row",
    width: "100%",
    [theme.breakpoints.down("xs")]: {
      marginTop: theme.spacing(3),
      marginBottom: theme.spacing(3),
    },

    [theme.breakpoints.up("lg")]: {
      flexDirection: "column",
      alignItems: "start",
    },
  },
  launchIcon: {
    fontSize: 20,
    marginLeft: "0.5rem",
    color: "#2f2f2f",
    "&:hover": {
      color: "#2f2f2f",
    },
  },
  reefRegionName: {
    marginBottom: "0.6rem",
  },
  cardTitleWrapper: {
    maxWidth: "calc(100% - 88px)",
    marginRight: theme.spacing(1),
  },
  cardTitle: {
    width: "100%",
    display: "block",
    overflow: "hidden",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
  },
}));

type SelectedReefContentProps = {
  reef: Reef;
  imageUrl?: string | null;
};

const SelectedReefContent = ({ reef, imageUrl }: SelectedReefContentProps) => {
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
  const sortedDailyData = sortByDate(reef.dailyData, "date");
  const dailyDataLen = sortedDailyData.length;
  const { maxBottomTemperature, satelliteTemperature, degreeHeatingDays } =
    (dailyDataLen && sortedDailyData[dailyDataLen - 1]) || {};

  const metrics = [
    {
      label: "SURFACE TEMP",
      value: formatNumber(satelliteTemperature, 1),
      unit: " °C",
      display: isNumber(satelliteTemperature),
    },
    {
      label: "HEAT STRESS",
      value: formatNumber(degreeHeatingWeeksCalculator(degreeHeatingDays), 1),
      unit: " DHW",
      display: isNumber(degreeHeatingDays),
    },
    {
      label: `TEMP AT ${reef.depth}m`,
      value: formatNumber(maxBottomTemperature, 1),
      unit: " °C",
      display: isNumber(maxBottomTemperature),
    },
  ];

  const { name, region: regionName } = getReefNameAndRegion(reef);

  const ChartComponent = (
    <Chart
      reefId={reef.id}
      dailyData={convertDailyDataToLocalTime(reef.dailyData, reef.timezone)}
      surveys={[]}
      temperatureThreshold={
        reef.maxMonthlyMean ? reef.maxMonthlyMean + 1 : null
      }
      maxMonthlyMean={reef.maxMonthlyMean || null}
      background
    />
  );

  return (
    <Grid
      className={
        imageUrl
          ? `${classes.cardWrapper} ${classes.mobileCardWrapperWithImage}`
          : `${classes.cardWrapper} ${classes.mobileCardWrapperWithNoImage}`
      }
      container
      justify="space-between"
      spacing={1}
    >
      {imageUrl && (
        <Grid item xs={12} sm={6} lg={4}>
          <Box position="relative" height="100%">
            <Link to={`/reefs/${reef.id}`}>
              <CardMedia className={classes.cardImage} image={imageUrl} />
            </Link>

            <Hidden smUp>
              <Box
                bgcolor="rgba(3, 48, 66, 0.75)"
                height="40%"
                width="100%"
                position="absolute"
                top={0}
                left={0}
              >
                <Box position="absolute" top={16} left={16}>
                  <Typography variant="h5">{name}</Typography>

                  {regionName && (
                    <Typography variant="h6" style={{ fontWeight: 400 }}>
                      {regionName}
                    </Typography>
                  )}
                </Box>
              </Box>
            </Hidden>

            <Box position="absolute" bottom={16} px={2} width="100%">
              <Grid
                container
                alignItems="center"
                justify={
                  reef.videoStream && isMobile ? "space-between" : "flex-end"
                }
              >
                {reef.videoStream && isMobile && (
                  <Chip
                    live
                    liveText="LIVE VIDEO"
                    to={`/reefs/${reef.id}`}
                    width={80}
                  />
                )}
                <Grid item>
                  <Link
                    style={{ color: "inherit", textDecoration: "none" }}
                    to={`/reefs/${reef.id}`}
                  >
                    <Button size="small" variant="contained" color="primary">
                      EXPLORE
                    </Button>
                  </Link>
                </Grid>
              </Grid>
            </Box>
          </Box>
        </Grid>
      )}

      <Grid
        item
        xs={12}
        sm={imageUrl ? 6 : 12}
        lg={imageUrl ? 6 : 10}
        style={{ marginBottom: "2rem", maxHeight: "14rem" }}
      >
        <Box pb="0.5rem" pl="0.5rem" pt="1.5rem" fontWeight={400}>
          <Hidden xsDown={Boolean(imageUrl)}>
            <Grid container alignItems="center">
              <Grid item className={classes.cardTitleWrapper}>
                <Typography
                  className={classes.cardTitle}
                  color="textSecondary"
                  variant="h5"
                >
                  <span title={name || ""}>{name}</span>
                </Typography>
              </Grid>
              {reef.videoStream && (
                <Chip
                  live
                  liveText="LIVE VIDEO"
                  to={`/reefs/${reef.id}`}
                  width={80}
                />
              )}
            </Grid>
            <Typography
              color="textSecondary"
              variant="h6"
              className={`${classes.cardTitle} ${classes.reefRegionName}`}
            >
              <span title={regionName || ""}>{regionName}</span>
            </Typography>
          </Hidden>
          <Typography color="textSecondary" variant="caption">
            DAILY SURFACE TEMP. (°C)
          </Typography>
        </Box>
        <Hidden xsDown>
          <div>{ChartComponent}</div>
        </Hidden>
        <Hidden smUp>{ChartComponent}</Hidden>
      </Grid>

      <Grid
        item
        xs={12}
        lg={2}
        style={{
          display: "flex",
        }}
      >
        <div className={classes.metricsContainer}>
          {metrics.map(({ label, value, unit, display }) => (
            <div key={label}>
              {display && (
                <>
                  <Typography variant="caption" color="textSecondary">
                    {label}
                  </Typography>
                  <Typography variant="h4" color="primary">
                    {value}
                    &nbsp;
                    <Typography variant="h6" component="span">
                      {unit}
                    </Typography>
                  </Typography>
                </>
              )}
            </div>
          ))}
        </div>
      </Grid>
    </Grid>
  );
};

SelectedReefContent.defaultProps = {
  imageUrl: null,
};

const featuredReefId = process.env.REACT_APP_FEATURED_REEF_ID || "";

const SelectedReefCard = () => {
  const classes = useStyles();
  const reef = useSelector(reefDetailsSelector);
  const loading = useSelector(reefLoadingSelector);
  const surveyList = useSelector(surveyListSelector);

  const isFeatured = `${reef?.id}` === featuredReefId;

  const { featuredSurveyMedia } =
    sortByDate(surveyList, "diveDate", "desc").find(
      (survey) =>
        survey.featuredSurveyMedia &&
        survey.featuredSurveyMedia.type === "image"
    ) || {};

  const hasMedia = Boolean(featuredSurveyMedia?.url);

  return (
    <Box className={classes.card}>
      {!loading && (
        <Box mb={2}>
          <Typography variant="h5" color="textSecondary">
            {isFeatured ? "Featured Site" : "Selected Site"}
            {!hasMedia && (
              <Link to={`reefs/${reef?.id}`}>
                <LaunchIcon className={classes.launchIcon} />
              </Link>
            )}
          </Typography>
        </Box>
      )}

      <Card>
        {loading ? (
          <Box
            height="20rem"
            display="flex"
            alignItems="center"
            justifyContent="center"
            textAlign="center"
            p={4}
          >
            <CircularProgress size="3rem" thickness={1} />
          </Box>
        ) : reef ? (
          <SelectedReefContent
            reef={reef}
            imageUrl={featuredSurveyMedia?.url}
          />
        ) : null}
      </Card>
    </Box>
  );
};

export default SelectedReefCard;
