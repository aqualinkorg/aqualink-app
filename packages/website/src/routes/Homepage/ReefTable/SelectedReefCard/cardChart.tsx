import React, { useState, useEffect, useCallback, useRef } from "react";
import { Line } from "react-chartjs-2";

import { createChartData } from "../../../../helpers/createChartData";
import { sortDailyData } from "../../../../helpers/sortDailyData";
import type { Data } from "../../../../store/Reefs/types";
import {
  createDatasets,
  calculateAxisLimits,
} from "../../../ReefRoutes/Reef/Charts/utils";

import "../../../../helpers/backgroundPlugin";
import "../../../../helpers/fillPlugin";

const CardChart = ({ dailyData, temperatureThreshold }: CardChartProps) => {
  const [updateChart, setUpdateChart] = useState<boolean>(false);
  const chartRef = useRef<Line>(null);
  const [xTickShift, setXTickShift] = useState<number>(0);

  // Sort daily data by date
  const sortByDate = sortDailyData(dailyData);

  const { surfaceTemperatureData } = createDatasets(sortByDate);

  const { xAxisMax, xAxisMin, chartLabels } = calculateAxisLimits(sortByDate);

  const changeXTickShift = () => {
    const { current } = chartRef;
    if (current) {
      const xScale = current.chartInstance.scales["x-axis-0"];
      const ticksPositions = xScale.ticks.map((_: any, index: number) =>
        xScale.getPixelForTick(index)
      );
      setXTickShift((ticksPositions[2] - ticksPositions[1]) / 2);
    }
  };

  const onResize = useCallback(() => {
    setUpdateChart(true);
    setTimeout(() => {
      setUpdateChart(false);
    }, 1);
  }, []);

  // Update chart and set temperature threshold position when window is resized
  useEffect(() => {
    window.addEventListener("resize", onResize);
    return () => {
      window.removeEventListener("resize", onResize);
    };
  }, [onResize]);

  useEffect(() => {
    changeXTickShift();
  });

  return (
    <Line
      ref={chartRef}
      options={{
        plugins: {
          chartJsPluginBarchartBackground: {
            color: "rgb(158, 166, 170, 0.07)",
            xTicksFontWeight: 2.8,
          },
          fillPlugin: {
            datasetIndex: 0,
            zeroLevel: temperatureThreshold,
            bottom: 0,
            top: 35,
            color: "rgba(250, 141, 0, 1)",
            updateChart,
          },
        },
        tooltips: {
          enabled: false,
        },
        legend: {
          display: false,
        },
        scales: {
          xAxes: [
            {
              type: "time",
              time: {
                displayFormats: {
                  hour: "MMM D h:mm a",
                },
                unit: "month",
              },
              ticks: {
                labelOffset: xTickShift,
                min: xAxisMin,
                max: xAxisMax,
                padding: 10,
                maxRotation: 0,
                callback: (value: string) => {
                  return value.split(", ")[0].toUpperCase();
                },
              },
              gridLines: {
                display: false,
                drawTicks: false,
              },
            },
          ],
          yAxes: [
            {
              gridLines: {
                drawTicks: false,
              },
              display: true,
              ticks: {
                min: 0,
                stepSize: 20,
                max: 40,
                callback: (value: number) => {
                  return `${value}\u00B0 `;
                },
              },
            },
          ],
        },
      }}
      height={100}
      data={createChartData(chartLabels, surfaceTemperatureData, 0, true)}
    />
  );
};

interface CardChartProps {
  dailyData: Data[];
  temperatureThreshold: number | null;
}

export default CardChart;
