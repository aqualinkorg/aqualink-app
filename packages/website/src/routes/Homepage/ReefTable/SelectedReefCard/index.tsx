import React from "react";
import {
  Typography,
  Paper,
  Grid,
  Button,
  CardMedia,
  withStyles,
  WithStyles,
  createStyles,
  Theme,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import CardChart from "./cardChart";
import { Reef } from "../../../../store/Reefs/types";
import { sortDailyData } from "../../../../helpers/sortDailyData";

import reefImage from "../../../../assets/reef-image.jpg";

const SelectedReefCard = ({ classes, reef }: SelectedReefCardProps) => {
  const sortByDate = sortDailyData(reef.dailyData);
  const dailyDataLen = sortByDate.length;

  return (
    <div className={classes.root}>
      <Typography
        style={{ margin: "0 0 0.5rem 1.5rem" }}
        variant="h5"
        color="textSecondary"
      >
        Featured Reef
      </Typography>
      <Grid container justify="center">
        <Paper elevation={3} className={classes.selectedReef}>
          <Grid className={classes.card} container item xs={12}>
            <Grid item xs={4}>
              <CardMedia className={classes.cardImage} image={reefImage} />
            </Grid>
            <Grid container item xs={6}>
              <Grid item xs={12}>
                <Typography
                  style={{ padding: "0.5rem 0 0 0.5rem" }}
                  color="textSecondary"
                  variant="h5"
                >
                  {reef.name}
                </Typography>
                <Typography
                  style={{ padding: "0 0 0.5rem 0.5rem", fontWeight: 400 }}
                  color="textSecondary"
                  variant="h6"
                >
                  {reef.region}
                </Typography>
              </Grid>
              <Grid item xs={12}>
                <Typography
                  style={{ padding: "0 0 0.5rem 0.5rem", fontWeight: 400 }}
                  color="textSecondary"
                  variant="subtitle1"
                >
                  MEAN DAILY WATER TEMPERATURE AT 25M (CÂ°)
                </Typography>
                <CardChart
                  dailyData={reef.dailyData}
                  temperatureThreshold={reef.temperatureThreshold}
                />
              </Grid>
            </Grid>
            <Grid
              style={{ paddingLeft: "2rem" }}
              container
              direction="row"
              alignItems="center"
              item
              xs={2}
            >
              <Grid item xs={12}>
                <Typography variant="caption" color="textSecondary">
                  TEMP AT 25M
                </Typography>
                <Typography
                  className={classes.cardMetrics}
                  variant="h4"
                  color="textSecondary"
                >
                  {sortByDate[dailyDataLen - 1].maxBottomTemperature}&#8451;
                </Typography>
              </Grid>
              <Grid item xs={12}>
                <Typography variant="caption" color="textSecondary">
                  SURFACE TEMP
                </Typography>
                <Typography
                  className={classes.cardMetrics}
                  variant="h4"
                  color="textSecondary"
                >
                  {sortByDate[dailyDataLen - 1].surfaceTemperature}&#8451;
                </Typography>
              </Grid>
              <Grid item xs={12}>
                <Typography variant="caption" color="textSecondary">
                  DEG. HEAT. DAYS
                </Typography>
                <Typography
                  className={classes.cardMetrics}
                  variant="h4"
                  color="textSecondary"
                >
                  {sortByDate[dailyDataLen - 1].degreeHeatingDays}
                </Typography>
              </Grid>
              <Grid item xs={12}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/reefs/${reef.id}`}
                >
                  <Button size="small" variant="contained" color="primary">
                    EXPLORE
                  </Button>
                </Link>
              </Grid>
            </Grid>
          </Grid>
        </Paper>
      </Grid>
    </div>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    root: {
      marginTop: "1rem",
    },
    selectedReef: {
      width: "48vw",
      height: "28vh",
    },
    card: {
      height: "100%",
    },
    cardImage: {
      borderRadius: "4px 0 0 4px",
      height: "100%",
    },
    cardMetrics: {
      color: theme.palette.primary.main,
    },
  });

interface selectedReefCardIncomingProps {
  reef: Reef;
}

type SelectedReefCardProps = selectedReefCardIncomingProps &
  WithStyles<typeof styles>;

export default withStyles(styles)(SelectedReefCard);
