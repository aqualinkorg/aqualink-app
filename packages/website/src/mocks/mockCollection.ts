import { CollectionDetails } from "../store/Collection/types";
import { mockReef } from "./mockReef";
import { mockUser } from "./mockUser";

export const mockCollection: CollectionDetails = {
  id: 1,
  isPublic: false,
  name: "Mock Collection",
  user: mockUser,
  reefs: [mockReef],
  reefIds: [1],
};
