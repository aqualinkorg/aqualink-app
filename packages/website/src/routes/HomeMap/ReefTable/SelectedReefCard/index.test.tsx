import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import SelectedReefCard from ".";
import { mockReef } from "../../../../mocks/mockReef";

jest.mock("react-chartjs-2", () => ({
  Line: () => "Mock-Line",
  Chart: {
    pluginService: {
      register: jest.fn(),
    },
  },
}));

const reef = {
  details: mockReef,
};

const mockStore = configureStore([]);

const store = mockStore({
  selectedReef: reef,
  homepage: {
    reefOnMap: reef,
  },
  surveyList: {
    list: [],
  },
  survey: {
    selectedSurvey: {
      details: null,
    },
  },
});

store.dispatch = jest.fn();

test("renders as expected", () => {
  process.env.REACT_APP_FEATURED_REEF_ID = "2";

  const { container } = render(
    <Provider store={store}>
      <Router>
        <SelectedReefCard />
      </Router>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});

test("renders loading as expected", () => {
  process.env.REACT_APP_FEATURED_REEF_ID = "4";
  const { container } = render(
    <Provider store={store}>
      <Router>
        <SelectedReefCard />
      </Router>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
