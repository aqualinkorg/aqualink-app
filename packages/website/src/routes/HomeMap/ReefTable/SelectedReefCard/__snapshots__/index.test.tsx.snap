// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-box
    classname="makeStyles-card-4"
  >
    <mock-box
      mb="2"
    >
      <mock-typography
        color="textSecondary"
        variant="h5"
      >
        Selected Site
        <a
          href="/reefs/1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-launchIcon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
      </mock-typography>
    </mock-box>
    <mock-card>
      <div
        class="MuiGrid-root makeStyles-cardWrapper-1 makeStyles-mobileCardWrapperWithNoImage-3 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-10"
          style="margin-bottom: 2rem; max-height: 14rem;"
        >
          <mock-box
            fontweight="400"
            pb="0.5rem"
            pl="0.5rem"
            pt="1.5rem"
          >
            <mock-hidden
              xsdown="false"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
              >
                <div
                  class="MuiGrid-root makeStyles-cardTitleWrapper-9 MuiGrid-item"
                >
                  <mock-typography
                    classname="makeStyles-cardTitle-10"
                    color="textSecondary"
                    variant="h5"
                  >
                    <span
                      title="Hawaii"
                    >
                      Hawaii
                    </span>
                  </mock-typography>
                </div>
              </div>
              <mock-typography
                classname="makeStyles-cardTitle-10 makeStyles-reefRegionName-8"
                color="textSecondary"
                variant="h6"
              >
                <span
                  title=""
                />
              </mock-typography>
            </mock-hidden>
            <mock-typography
              color="textSecondary"
              variant="caption"
            >
              DAILY SURFACE TEMP. (°C)
            </mock-typography>
          </mock-box>
          <mock-hidden
            xsdown="true"
          >
            <div>
              Mock-Line
            </div>
          </mock-hidden>
          <mock-hidden
            smup="true"
          >
            Mock-Line
          </mock-hidden>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-2"
          style="display: flex;"
        >
          <div
            class="makeStyles-metricsContainer-6"
          >
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                SURFACE TEMP
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                23.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                HEAT STRESS
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                4.9
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   DHW
                </mock-typography>
              </mock-typography>
            </div>
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                TEMP AT 0m
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                39.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
          </div>
        </div>
      </div>
    </mock-card>
  </mock-box>
</div>
`;

exports[`renders loading as expected 1`] = `
<div>
  <mock-box
    classname="makeStyles-card-15"
  >
    <mock-box
      mb="2"
    >
      <mock-typography
        color="textSecondary"
        variant="h5"
      >
        Selected Site
        <a
          href="/reefs/1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-launchIcon-18"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
      </mock-typography>
    </mock-box>
    <mock-card>
      <div
        class="MuiGrid-root makeStyles-cardWrapper-12 makeStyles-mobileCardWrapperWithNoImage-14 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-10"
          style="margin-bottom: 2rem; max-height: 14rem;"
        >
          <mock-box
            fontweight="400"
            pb="0.5rem"
            pl="0.5rem"
            pt="1.5rem"
          >
            <mock-hidden
              xsdown="false"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
              >
                <div
                  class="MuiGrid-root makeStyles-cardTitleWrapper-20 MuiGrid-item"
                >
                  <mock-typography
                    classname="makeStyles-cardTitle-21"
                    color="textSecondary"
                    variant="h5"
                  >
                    <span
                      title="Hawaii"
                    >
                      Hawaii
                    </span>
                  </mock-typography>
                </div>
              </div>
              <mock-typography
                classname="makeStyles-cardTitle-21 makeStyles-reefRegionName-19"
                color="textSecondary"
                variant="h6"
              >
                <span
                  title=""
                />
              </mock-typography>
            </mock-hidden>
            <mock-typography
              color="textSecondary"
              variant="caption"
            >
              DAILY SURFACE TEMP. (°C)
            </mock-typography>
          </mock-box>
          <mock-hidden
            xsdown="true"
          >
            <div>
              Mock-Line
            </div>
          </mock-hidden>
          <mock-hidden
            smup="true"
          >
            Mock-Line
          </mock-hidden>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-2"
          style="display: flex;"
        >
          <div
            class="makeStyles-metricsContainer-17"
          >
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                SURFACE TEMP
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                23.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                HEAT STRESS
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                4.9
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   DHW
                </mock-typography>
              </mock-typography>
            </div>
            <div>
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                TEMP AT 0m
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                39.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
          </div>
        </div>
      </div>
    </mock-card>
  </mock-box>
</div>
`;
